dvc/cli.py

```diff
@@ -154,18 +274,61 @@ def get_parent_parser():
     return parent_parser
 
 
+def add_parser_exit_on_error(self, name, **kwargs):
```

Comment:
Btw, `exit_on_error` calls `error` method of the parser, which is overriden in `DvcParser`, so seems like it shouldn't exit right now. Or am I missing something?

Reply:
I'm not sure I follow. `exit_on_error` calls `error` method of the parser if it is set to `True`.

```python
# parse the arguments and exit if there are any errors
if self.exit_on_error:
    try:
        namespace, args = self._parse_known_args(args, namespace)
    except ArgumentError:
        err = _sys.exc_info()[1]
        self.error(str(err))
else:
    namespace, args = self._parse_known_args(args, namespace)
```

https://github.com/python/cpython/blob/master/Lib/argparse.py#L1849

`logger.error` is then called in the `_find_cmd_suggestions` method when displaying command suggestions.

Reply:
You don't need exit_on_error at all, our `DvcParser.error()` will not sys.exit, just catch DvcParserError instead of ArgumentError